hy/compiler.py

```diff
@@ -2035,7 +2049,9 @@ def compile_return(self, expr):
     def compile_class_expression(self, expressions):
         def rewire_init(expr):
             new_args = []
-            if expr[0] == HySymbol("setv"):
+            if (isinstance(expr, HyExpression)
+                and len(expr) > 1
+                and expr[0] == HySymbol("setv")):
```

Comment:
This will trigger on the string literal `"setv"` without #1504.

Reply:
Put an `isinstance` check. Don't have the time at the moment to dig into Hy stuff (very soon though), but might as well try and get these bug fixes in as things work currently.